iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_receiverRuntime{&iox::runtime::PoshRuntime::getInstance("/receiver")};
+    PoshRuntime* m_senderRuntime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::CString100 m_runnableName = iox::cxx::CString100("testRunnable");
+    const iox::cxx::CString100 m_invalidRunnableName = iox::cxx::CString100("invalidRunnable,");
+    static bool m_errorHandlerCalled;
+};
+
+bool PoshRuntime_test::m_errorHandlerCalled{false};
+
+
+TEST_F(PoshRuntime_test, AppnameLength_TooLong)
+{
+    const std::string string104chars =
+        "/MXIYXHyPF9KjXAPv9ev9jxofYDArZzTvf8FF5uaWWC4dwabcjW75DurqeN645IabAsXVfngor7784446vb4vhArwBxLZlN1k1Qmrtz";
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(string104chars); },
+                 "Application name has more than 100 characters, including null termination!");
+}
+
+
+TEST_F(PoshRuntime_test, AppnameLength_ok)
+{
+    const std::string string100chars =
+        "/MXIYXHyPF9KjXAPv9ev9jxofYDArZzTvf8FF5uaWWC4dwabcjW75DurqeN645IabAsXVfngor7784446vb4vhArwBxLZlN1k1";
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(string100chars); });
+}
+
+
+TEST_F(PoshRuntime_test, NoAppname)
+{
+    const std::string wrong("");
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name must not be empty!");
+}
+
+
+TEST_F(PoshRuntime_test, NoLeadingSlash_Appname)
+{
+    const std::string wrong = "wrongname";
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name wrongname does not have the required leading slash '/'");
+}
+
+
+// test class creates instance of Poshruntime, so when getInstance() is called without name it reuturns extisting
+// instance
+TEST_F(PoshRuntime_test, DISABLED_AppnameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceName)
+{
+    const std::string appname = "/app";
+
+    auto& sut = PoshRuntime::getInstance(appname);
+
+    EXPECT_EQ(sut.getInstanceName(), appname);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ReturnValue)
+{
+    uint32_t uniqueIdCounter = iox::popo::BasePortData::s_uniqueIdCounter;
+
+    const auto applicationPortData = m_senderRuntime->getMiddlewareApplication();
+
+    EXPECT_EQ(std::string("/sender"), applicationPortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), applicationPortData->m_serviceDescription);
+    EXPECT_EQ(false, applicationPortData->m_toBeDestroyed);
+    EXPECT_EQ(uniqueIdCounter, applicationPortData->m_uniqueId);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ApplicationlistOverflow)
+{
+    m_errorHandlerCalled = false;
```

Comment:
Could be done in the setup code as well (but will then be done every time, so probably just do it when needed).

Reply:
If it's done this way, we should always check that the error handler was not called in all the testcases where we do not expect it to be called.
By changing the error handler for all tests, this is necessary since a test might have triggered the error handler but since it doesn't do anything except setting the flag, a situation which might lead to a termination in real word usage might result in a passed test by accident.